app/server/tests/api-user.test.js

```diff
@@ -0,0 +1,243 @@
+
+    test('User is successfully found - returns 200', () => {
+        expect(response.status).toBe(200);
+    });
+
+    // This is only returning name and email, despite what Swagger suggests.
+    test('Body of response is formated as expected (object with certain keys)', () => {
+        expect(typeof response.body).toBe('object');
+        expect(response.body.name).toBeTruthy();
+        expect(response.body.email).toBeTruthy();
+        // expect(response.body.first_name).toBeTruthy();
+        // expect(response.body.last_name).toBeTruthy();
+        // expect(response.body.bio).toBeTruthy();
+        // expect(response.body.title).toBeTruthy();
+    });
+});
+
+// This needs fixing, needs another account to get data from
+describe('Get information of other users with /user/{name}', () => {
+    let userName = name.gen();
+    let userPassword = password.gen();
+    let loginResponse;
+    let response; 
+    beforeAll(async () => {
+        await auth.register(userName, email.gen(), userPassword);
+        loginResponse = await auth.login(userName, userPassword);
+        response = await user.getUserByName(loginResponse.body.token, userName);
+    });
+
+    afterAll(async () => {
+        await auth.deleteUsers();
+    });
+
+    test('User is successfully found - returns 200', () => {
+        expect(response.status).toBe(200);
+    });
+
+    test('Body of response is formated as expected (object with certain keys, but not all keys)', () => {
+        expect(typeof response.body).toBe('object');
+        expect(response.body.name).toBeTruthy();
+        expect(response.body.email).toBeTruthy();
+        expect(response.body.first_name).not.toBeTruthy();
+        expect(response.body.last_name).not.toBeTruthy();
+        expect(response.body.bio).not.toBeTruthy();
+        expect(response.body.title).not.toBeTruthy();
+    });
+
+    test('Trying to get a user with a bad token rejects the request - returns 403', async () => {
+        let tempResponse = await user.getUserByName('thistokenisillegitimate', userName);
+        expect(tempResponse.status).toBe(403);
+    });
+
+    test('Trying to get data of another user should only return name and email', async () => {
+        await auth.register('Todd', 'todd@gmail.com', 'Todd123!'); // Create new user
+        let tempResponse = await user.getUserByName(loginResponse.body.token, 'Todd'); // Try to get that user's info with test account's token.
+        expect(tempResponse.status).toBe(200);
+        expect(typeof tempResponse.body).toBe('object');
+        expect(tempResponse.body.name).toBeTruthy();
+        expect(tempResponse.body.email).toBeTruthy();
+        expect(tempResponse.body.first_name).not.toBeTruthy();
+        expect(tempResponse.body.last_name).not.toBeTruthy();
+        expect(tempResponse.body.bio).not.toBeTruthy();
+        expect(tempResponse.body.title).not.toBeTruthy();
+    });
+});
+
+describe('Edit the information of users with /user/{name}', () => {
```

Comment:
I removed the ability to edit user.name

Reply:
There's one test there that tries to change one's own name. I've edited the description to make that more clear.